great_expectations/experimental/datasources/interfaces.py

```diff
@@ -364,7 +367,7 @@ def _load_asset_subtype(
         """
         logger.info(f"Loading '{data_asset.name}' asset ->\n{pf(data_asset, depth=4)}")
         asset_type_name: str = data_asset.type
-        asset_type: Type[_DataAssetT] = _SourceFactories.type_lookup[asset_type_name]
+        asset_type: Type[_DataAssetT] = cls._type_lookup[asset_type_name]
```

Comment:
The `TypeLookup` for the assets is now on the `DataSource` class rather than the shared `_SourceFactories.type_lookup`